borg/helpers.py

```diff
@@ -151,7 +152,13 @@ def load(cls, repository, key=None, force_tam_not_required=False):
     def write(self):
         if self.key.tam_required:
             self.config[b'tam_required'] = True
-        self.timestamp = datetime.utcnow().isoformat()
+        # self.timestamp needs to be strictly monotonically increasing. Clocks often are not set correctly
+        if self.timestamp is None:
+            self.timestamp = datetime.utcnow().isoformat()
+        else:
+            prev_ts = datetime.strptime(self.timestamp, "%Y-%m-%dT%H:%M:%S.%f")
+            incremented = (prev_ts + timedelta(microseconds=1)).isoformat()
+            self.timestamp = max(incremented, datetime.utcnow().isoformat())
```

Comment:
To actually provide rollback prevention it needs to be higher than the previous one. To just silence the warning it might not.
I somehow thought that the cache also uses the manifest timestamp, but it does use the id which is partly derived from the timestamp (the id beeing the id_hash of the whole manifest). I feel a lot safer if the manifest has a component that will never be the same twice. And we plan to move the TAM out into some kind of envelop so we should‘t leave a footgun laying around depending on the TAM. 

That said, not having to parse the date would be nice. But i don‘t trust that construction.